File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    for (int i = 0; i < s.size(); i++)
    {
        if (i == 0)
            answer.push_back(-1);
        else
        {
            int temp = i;
            int count = 0;
            while (temp--)
            {
                if (s[i] == s[temp])
                    break;
                count++;
            }
            if (temp < 0 && s[i] != s[0])
                answer.push_back(-1);
            else if (temp >= 0)
                answer.push_back(count + 1);
        }
    }
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] solution(String s) {
        Map<Character, Integer> map = new HashMap<>();

        int[] answer = new int[s.length()];
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i++) {
            if (!map.containsKey(chars[i])) {
                map.put(chars[i], i);
                answer[i] = -1;
            } else {
                int idx = map.get(chars[i]);
                answer[i] = i - idx;
                map.put(chars[i], i);
            }
        }
        
        
        return answer;
    }
}